### PR TITLE
Unfuck bash into Golang

### PR DESCRIPTION
**Pull request by slavaaaaaaaaaa**
- [x] Step 1: convert crappy gamedata to FEN:
   - [x] piece placement from white's perspective 
   - [x] active colour
   - [x] castling availability
   - [x] En passant target square
   - [x] halfmove clock
   - [x] fullmove number
- Step 2: wrap API calls in golang (another PR)
- Step 3: wrap stockfish UCI with golang (another PR)

### CODE REVIEW

**Comment by asciifaceman** [comment]
back up and punt with cobra (cobra init, cobra add, etc) 

https://github.com/spf13/cobra

you will thank me

File: chesstime.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+        s := square.Parse(strings.Split(pos, "=")[0])
+        p := piece.New(colourmap[string(pos[3])], piecemap[string(pos[4])])
+
+        bored.Put(p, s)
+    }
+
+    fmt.Println(bored)
+
+    encoded, _ := fen.Encode(bored) // TODO: how in the world can i do both of these in one line?
```
Comment (asciifaceman) [inline chesstime.go]: what do you mean?

Comment (smaslennikov) [reply chesstime.go]: `fen.Encode()` returns two values: I only want the first one. What I want to do is combine this variable assignment with the return statement on the next line.

Comment (asciifaceman) [reply chesstime.go]: oh yeah, well `_` in go is literally just used for that lol

but you shouldn't execute things on the return line anyways

File: cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
+// Copyright © 2018 OrcaLLC
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
```
Comment (asciifaceman) [inline cmd/root.go]: we should specify that this can be changed without notice if our pockets are light

**Review by asciifaceman — APPROVED** [review]

**Comment by smaslennikov** [comment]
This should probably be licensed under JaveLic

**Comment by smaslennikov** [comment]
@asciifaceman @taup1n I'm going to merge this as the feature is complete; but would you mind leaving comments on anything odd you see anyway, so I can fix it in post?

File: cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    // has an action associated with it:
+    //    Run: func(cmd *cobra.Command, args []string) { },
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+    if err := rootCmd.Execute(); err != nil {
+        fmt.Println(err)
+        os.Exit(1)
```
Comment (asciifaceman) [inline cmd/root.go:34]: could use a panic() instead, look up "exiting golang application"

File: cmd/root.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+func initConfig() {
+    if cfgFile != "" {
+        // Use config file from the flag.
+        viper.SetConfigFile(cfgFile)
+    } else {
+        // Find home directory.
+        home, err := homedir.Dir()
+        if err != nil {
+            fmt.Println(err)
+            os.Exit(1)
```
Comment (asciifaceman) [inline cmd/root.go:61]: same as above

File: cmd/root.go (hunk 3 above)
Comment (slavaaaaaaaaaa) [reply cmd/root.go:34]: Oh this was generated with cobra.

File: cmd/root.go (hunk 4 above)
Comment (slavaaaaaaaaaa) [reply cmd/root.go:61]: this too

**Pull request merged**